lib/matplotlib/backends/_backend_tk.py

```diff
@@ -234,10 +233,13 @@ def filter_destroy(event):
         self._rubberband_rect_white = None
 
     def _update_device_pixel_ratio(self, event=None):
-        # Tk gives scaling with respect to 72 DPI, but Windows screens are
-        # scaled vs 96 dpi, and pixel ratio settings are given in whole
-        # percentages, so round to 2 digits.
-        ratio = round(self._tkcanvas.tk.call('tk', 'scaling') / (96 / 72), 2)
+        if sys.platform == 'win32':
+            # Tk gives scaling with respect to 72 DPI, but Windows screens are
+            # scaled vs 96 dpi, and pixel ratio settings are given in whole
+            # percentages, so round to 2 digits.
+            ratio = round(self._tkcanvas.tk.call('tk', 'scaling') / (96 / 72), 2)
+        else:
```

Comment:
```suggestion
        elif sys.platform == "linux":
```

Reply:
Thank you for testing it out! I limited the change to just "linux" and made sure `_set_device_pixel_ratio` is called only on supported platforms.